src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -27,243 +32,218 @@
+        per country. This function retrieves the country data
+        if it exists, and else the regional data for `date_pub`.
+
+        Parameters
+        ----------
+        date_pub : date or str
+            date for which the data should be downloaded
+            only the year and month part are used
+            this commonly refers to the month of the Current Situation period
+            if str, it should be in ISO 8601 format, e.g. '2021-10-01'
+        clobber : bool, default = False
+            If True, overwrites existing raw files
+
+        Examples
+        --------
+        >>> download(date_pub="2020-10-01")
+        """
+        # convert to datetime if str
+        if not isinstance(date_pub, date):
+            date_pub = datetime.strptime(date_pub, _DATE_FORMAT)
+
+        # we prefer the country data as this more nicely structured
+        # thus first check if that is available
+        country_path = self._download_country(
+            date_pub=date_pub,
+            clobber=clobber,
+        )
+        if country_path is not None:
+            return country_path
+        else:
+            region_path = self._download_region(
+                date_pub=date_pub,
+                clobber=clobber,
+            )
+        if region_path is not None:
+            return region_path
+        else:
+            raise RuntimeError(
+                f"No data found for {date_pub.strftime('%Y-%m')}"
+            )
 
-def _download_zip(
-    url: str,
-    zip_filename: str,
-    output_dir: Path,
-) -> Optional[Path]:
-    """
-    Download and unzip the file at the url.
+    def _download_country(
+        self,
+        date_pub: date,
+        clobber: bool,
+    ) -> Optional[Path]:
+        """
+        Download fewsnet data that covers the iso2 country.
+
+        Returns
+        -------
+        country_data : str
+            if data found return the output_dir, else return None
+        """
+        url_country_date = _BASE_URL_COUNTRY.format(
+            iso2=self._iso2, YYYY=date_pub.year, MM=f"{date_pub.month:02d}"
+        )
 
-    Parameters
-    ----------
-    url : str
-        url that contains the zip file to be downloaded
-    zip_filename : str
-        name of the zipfile
-    output_dir : Path
-        path of dir to which the zip content should be written
+        # filenames have upper iso2, so use that for dirs as well
+        output_dir_country = (
+            self._raw_base_dir / f"{self._iso2}{date_pub.strftime('%Y%m')}"
+        )
+        zip_filename = f"{output_dir_country.name}.zip"
+        if not output_dir_country.exists() or clobber is True:
+            country_data = self._download_zip(
+                url=url_country_date,
+                zip_filename=zip_filename,
+                output_dir=output_dir_country,
+            )
+        else:
+            country_data = output_dir_country
+        return country_data
+
+    def _download_region(
```

Comment:
There's a lot of overlap with `_download_country`, maybe they can be refactored a bit?